src/app/posts/models/Post.model.ts

```diff
@@ -0,0 +1,20 @@
+import {Author} from "./Author.model";
+
+export class Post {
+  constructor(
+    public id: string,
+    public author: Author,
+    public title: string,
+    public text: string,
+    public likes: number,
+    public date: Date) {
+  }
+}
+
+export class CreatePostPartialData {
+  constructor(
+    public title: string,
+    public text: string,
+  ) {
+  }
+}
```

Comment:
You can create interface Post
```
export class CreatePostPartialModel implements Partial<Post>{}
or use Pick
export class PostModel implements Post{ }
```

CreatePost and Post should be linked(inheritance, composition)